.github/workflows/build_workflow.yml

```diff
@@ -51,11 +51,11 @@ jobs:
         uses: docker/build-push-action@v5
         with:
           context: .
-          push: ${{ github.event_name != 'pull_request' }}
+          push: ${{ github.event_name != 'pull_request' || github.event.pull_request.merged }}
```

Comment:
new docker image will be published if event is not pull_request or if it's pull_request.merged